Add Fleet & Agent 8.17.0 Release Notes (#1498)
elastic/ingest-docs: docs/en/ingest-management/release-notes/release-notes-8.17.asciidoc

:kibana-issue: https://github.com/elastic/kibana/issues/
:kibana-pull: https://github.com/elastic/kibana/pull/
:beats-issue: https://github.com/elastic/beats/issues/
:beats-pull: https://github.com/elastic/beats/pull/
:agent-libs-pull: https://github.com/elastic/elastic-agent-libs/pull/
:agent-issue: https://github.com/elastic/elastic-agent/issues/
:agent-pull: https://github.com/elastic/elastic-agent/pull/
:fleet-server-issue: https://github.com/elastic/fleet-server/issues/
:fleet-server-pull: https://github.com/elastic/fleet-server/pull/

[[release-notes]]
= Release notes

This section summarizes the changes in each release.

* <<release-notes-8.17.0>>

Also see:

* {kibana-ref}/release-notes.html[{kib} release notes]
* {beats-ref}/release-notes.html[{beats} release notes]

// begin 8.17.0 relnotes

[[release-notes-8.17.0]]
== {fleet} and {agent} 8.17.0

Review important information about the {fleet} and {agent} 8.17.0 release.

[discrete]
[[breaking-changes-8.17.0]]
=== Breaking changes

Breaking changes can prevent your application from optimal operation and
performance. Before you upgrade, review the breaking changes, then mitigate the
impact to your application.

{agent}::

> * Install images for {agent} on Debian 10 systems are no longer being made available. Debian 11 is now supported instead. {agent-pull}5847[#5847]

Reviewer: ```suggestion
* {agent} is now compiled using Debian 11 and linked against glibc 2.31 instead of 2.19. Drops support for Debian 10. {agent-pull}5847[#5847]
```

[discrete]
[[new-features-8.17.0]]
=== New features

The 8.17.0 release Added the following new and notable features.

{fleet}::
* Expose advanced file logging configuration in the UI. {kibana-pull}200274[#200274]

{agent}::

> * Add support for pre-existing Active Directory user for unprivileged mode. {agent-pull}5988[#5988] {agent-issue}4585[#4585]

Reviewer: ```suggestion
* Add support for running as a pre-existing user when installing in unprivileged mode, with technical preview support for pre-existing Windows Active Directory users. {agent-pull}5988[#5988] {agent-issue}4585[#4585]
```

> * Add support for the Filestream integration. This will enable migration from the custom log integration which is based on a log input that is planned for deprecation. https://github.com/elastic/integrations/pull/11332[#11332]

Author: This one doesn't appear in the agent changelog, since the change is from the integration repo, but it is included in Nima's 8.17 feature release summary, so I've included it.

> * Add support for the Filestream integration. This will enable migration from the custom log integration which is based on a log input that is planned for deprecation. https://github.com/elastic/integrations/pull/11332[#11332]

Reviewer: ```suggestion
* Add a new custom [Filestream logs integration](https://github.com/elastic/integrations/tree/main/packages/filestream). This will enable migration from the custom log integration which is based on a log input that is planned for deprecation. https://github.com/elastic/integrations/pull/11332[#11332]. 
```

Author: Updated the comment so the link is in Asciidoc format.

[discrete]
[[enhancements-8.17.0]]
=== Enhancements

{fleet}::
* Filter the integrations/packages list shown in the UI depending on the `policy_templates_behavior` field. {kibana-pull}200605[#200605]
* Add a `<type>@custom` component template to integrations index template's `composed_of` array. {kibana-pull}192731[#192731]

{fleet-server}::
* Update `elastic-agent-libs` to version `0.14.0`. {fleet-server-pull}4042[#4042]

{agent}::
* Enable persistence in the configuration provided with our OTel Collector distribution. {agent-pull}5549[#5549]

> * Add Filebeat as a receiver for {agent} running in OTel mode. {agent-pull}5833[#5833]

Reviewer: ```suggestion
```

We aren't ready for anyone to use this yet.

> * Restrict using the CLI to upgrade {fleet}-managed {agents}. With this implementation, upgrading a {fleet}-managed agent using the CLI is supported only if the agent is running in privileged mode and the user provides a `force` flag. {agent-pull}5864[#5864] {agent-issue}4890[#4890]

Reviewer: ```suggestion
* Restrict using the CLI to upgrade for {fleet}-managed {agents}. {agent-pull}5864[#5864] {agent-issue}4890[#4890]
```

We don't actually want people to do this and the `--force` flag is hidden as an escape hatch.

* Add `os_family`, `os_platform` and `os_version` to the {agent} host provider, enabling differentiating Linux distributions. This is required to support Debian 12 and other distributions that are moving away from traditional log files in favour of Journald. {agent-pull}5941[#5941] https://github.com/elastic/integrations/issues/10797[10797] https://github.com/elastic/integrations/pull/11618[11618]
* Emit Pod data only for Pods that are running in the Kubernetes provider. {agent-pull}6011[#6011] {agent-issue}5835[#5835] {agent-issue}5991[#5991] 
* Remove {endpoint-sec} from Linux container images. {endpoint-sec} cannot run in containers since it has a systemd dependency. {agent-pull}6016[#6016] {agent-issue}5495[#5495]

> * Update the outdated Kubernetes composable provider README. {agent-pull}5990[#5990]

Reviewer: ```suggestion
```

Not a user facing change.
